### PR TITLE
[TEP-0145] Pull in cel-go to vendor

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This is the first step of [TEP-0145](https://github.com/tektoncd/community/blob/main/teps/0145-cel-in-whenexpression.md) tracking issue is https://github.com/tektoncd/pipeline/issues/7244, same as [tekton/triggers](https://github.com/tektoncd/triggers/blob/main/pkg/interceptors/cel/cel.go#L29), we will use cel-go as the library to evaluate the CEL expression. 

This PR pulls the dependency first to avoid rebasing all the time and help reviewers to better review the following feature PRs by reducing noises, no functional change in this PR

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/pipeline/v1/when_validation.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"fmt"
 	"strings"
 
+	_ "github.com/google/cel-go/cel"
```
Comment (QuanZhang-William) [inline pkg/apis/pipeline/v1/when_validation.go:23]: Maybe add a todo comment here?

**Review by QuanZhang-William — COMMENTED** [review]
/lgtm

File: pkg/apis/pipeline/v1/when_validation.go (hunk 1 above)
Comment (Yongxuanzhang) [reply pkg/apis/pipeline/v1/when_validation.go:24]: Thanks! Added a todo linking to the tracking issue

**Review by jerop — APPROVED** [review]

**Pull request merged**